protostar/configuration_file/configuration_toml_reader.py

```diff
@@ -0,0 +1,113 @@
+from pathlib import Path
+from typing import Any, Dict, List, Optional, cast
+
+import flatdict
+import tomli
+
+from protostar.protostar_exception import ProtostarException
+
+
+class ConfigurationTOMLReader:
+    QualifiedSectionName = str
+
+    def __init__(self, path: Path, ignore_attribute_casing: bool = False):
```

Comment:
This class contains a logic for reading a file, which in my opinion is a bad thing and you can already see effects of this in tests because you have to create a physical file. Please read about https://docs.python.org/3/library/io.html module and abstract away IO operations here by taking a TextIOBase object.

Alternatively, I think we should be fine by simply accepting file contents here! Our configuration files are **never** going to be huge enough to have us to worry about them taking terrible amounts of memory.

Reply:
Changed `ConfigurationTOMLInterpreter` (old `Reader`) to accept string instead of path.